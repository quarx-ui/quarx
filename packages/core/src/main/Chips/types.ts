import { ReactElement } from 'react';
import { BaseProps, ComponentPropsWithHTML, SVGIcon, Values, WithClassesAndStyles } from '@core';
import { CHIPS_VARIANT } from './constants';
import { ChipsCSSVarKeys, ChipsStyleKeys, ChipsStyleParams } from './styles';

export type ChipsVariant = Values<typeof CHIPS_VARIANT>;

export interface ChipsPropsWithoutHtml extends
    Partial<Omit<ChipsStyleParams, 'leftIconExists' | 'rightIconExists'>>,
    WithClassesAndStyles<ChipsStyleKeys, ChipsStyleParams, ChipsCSSVarKeys>,
    BaseProps<HTMLButtonElement>
{
    /** @description Текст */
    children?: string;

    /** @description Иконка, отображаемая слева в активированном состоянии
     *
     * @default CheckMarkIcon */
    activeStateIcon?: ReactElement | SVGIcon;

    /** @description Левая иконка */
    leftIcon?: ReactElement | SVGIcon;

    /** @description Правая иконка (кастомная либо стандартная, если передан "true") */
    rightIcon?: boolean | ReactElement | SVGIcon;

    /** @description Скрыть элемент
     *
     * @default false */
    hidden?: boolean;
}

export type ChipsProps = ComponentPropsWithHTML<ChipsPropsWithoutHtml, 'button'>;

export * from './styles/types';
