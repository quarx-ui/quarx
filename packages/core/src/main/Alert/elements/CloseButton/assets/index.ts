import { ReactComponent as CrossIconSmall } from './cross-small.svg';
import { ReactComponent as CrossIcon } from './cross.svg';

export {
    CrossIconSmall,
    CrossIcon,
};
