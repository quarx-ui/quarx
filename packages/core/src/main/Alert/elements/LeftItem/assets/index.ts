import { ReactComponent as LightningIcon } from './lightning.svg';
import { ReactComponent as EnvelopeIconSmall } from './envelope-small.svg';
import { ReactComponent as InfoIcon } from './info.svg';
import { ReactComponent as CheckmarkCircleIconSmall } from './checkmark-circle-small.svg';
import { ReactComponent as AttentionTriangleIconSmall } from './attention-triangle-small.svg';
import { ReactComponent as AttentionRhombusIconSmall } from './attention-rhombus-small.svg';
import { ReactComponent as DocumentTextIcon } from './document-text.svg';
import { ReactComponent as EnvelopeIconLarge } from './envelope-large.svg';
import { ReactComponent as AttentionCircleIcon } from './attention-circle.svg';
import { ReactComponent as CheckmarkCircleIconLarge } from './checkmark-circle-large.svg';
import { ReactComponent as AttentionTriangleIconLarge } from './attention-triangle-large.svg';
import { ReactComponent as AttentionRhombusIconLarge } from './attention-rhombus-large.svg';

export {
    LightningIcon,
    EnvelopeIconSmall,
    InfoIcon,
    CheckmarkCircleIconSmall,
    AttentionTriangleIconSmall,
    AttentionRhombusIconSmall,
    DocumentTextIcon,
    EnvelopeIconLarge,
    AttentionCircleIcon,
    CheckmarkCircleIconLarge,
    AttentionTriangleIconLarge,
    AttentionRhombusIconLarge,
};
