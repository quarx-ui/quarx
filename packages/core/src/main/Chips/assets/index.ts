import { FC, SVGProps } from 'react';
import { ChipsSize } from '@core';
import { ReactComponent as CheckMarkIconSmallSize } from './small/checkmarkIcon.svg';
import { ReactComponent as CheckMarkIconMediumSize } from './medium/checkmarkIcon.svg';
import { ReactComponent as ClosingIconSmallSize } from './small/closeIcon.svg';
import { ReactComponent as ClosingIconMediumSize } from './medium/closeIcon.svg';
import { ReactComponent as TriangleIconSmallSize } from './small/triangleIcon.svg';
import { ReactComponent as TriangleIconMediumSize } from './medium/triangleIcon.svg';

export const CheckMarkIcon: Record<ChipsSize, FC<SVGProps<SVGElement>>> = {
    small: CheckMarkIconSmallSize,
    medium: CheckMarkIconMediumSize,
};

export const ClosingIcon: Record<ChipsSize, FC<SVGProps<SVGElement>>> = {
    small: ClosingIconSmallSize,
    medium: ClosingIconMediumSize,
};

export const TriangleIcon: Record<ChipsSize, FC<SVGProps<SVGElement>>> = {
    small: TriangleIconSmallSize,
    medium: TriangleIconMediumSize,
};
