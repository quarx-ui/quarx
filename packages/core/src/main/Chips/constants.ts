import { valuesAsKeysFromArray } from '@core';

export const CHIPS_VARIANT = valuesAsKeysFromArray([
    'input',
    'filter',
]);
