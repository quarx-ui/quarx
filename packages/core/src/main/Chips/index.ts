export { Chips } from './Chips';
export * from './constants';
export * from './types';
