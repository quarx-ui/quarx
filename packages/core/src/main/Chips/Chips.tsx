/** @jsx jsx */
import { jsx } from '@emotion/react';
import React, { FC, forwardRef, useRef } from 'react';
import { transitions, usePropsOverwrites } from '@core/styles';
import { QX_SIZE } from '@core/enums';
import { CheckMarkIcon, ClosingIcon, TriangleIcon } from '@core/src/main/Chips/assets';
import { SwitchTransition } from 'react-transition-group';
import { focusable, If, Transition } from '@core';
import { useId } from '@core/utils';
import { CHIPS_VARIANT } from './constants';
import { ChipsProps } from './types';
import { CHIPS_CSS_VARS, useStyles } from './styles';

const mapVariantToRightIcon = {
    [CHIPS_VARIANT.input]: ClosingIcon,
    [CHIPS_VARIANT.filter]: TriangleIcon,
};

export const Chips: FC<ChipsProps> = forwardRef<HTMLButtonElement, ChipsProps>((
    initialProps,
    ref,
) => {
    const { cn, props, styleProps } = usePropsOverwrites('Chips', initialProps, CHIPS_CSS_VARS);
    const {
        children,
        variant = CHIPS_VARIANT.input,
        active,
        elevation,
        disabled,
        size = QX_SIZE.medium,
        disableFocus,
        leftIcon,
        leftIconColor,
        rightIcon,
        activeStateIcon,
        rotateRightIcon = true,
        hidden,
        ...restProps
    } = props;

    const rightIconExists = Boolean(rightIcon);
    const RightIcon = React.isValidElement(rightIcon)
        ? rightIcon
        : React.createElement(mapVariantToRightIcon[variant][size]);

    const ActiveStateIcon = React.isValidElement(activeStateIcon)
        ? activeStateIcon
        : React.createElement(CheckMarkIcon[size]);
    const LeftIcon = active
        ? ActiveStateIcon
        : React.isValidElement(leftIcon) && leftIcon;
    const leftIconExists = Boolean(LeftIcon);

    const inactiveLeftIconRef = useRef<HTMLDivElement>(null);
    const activeLeftIconRef = useRef<HTMLDivElement>(null);
    const inactiveLeftIconId = useId();
    const activeLeftIconId = useId();
    const leftIconId = active ? activeLeftIconId : inactiveLeftIconId;
    const leftIconRef = active ? activeLeftIconRef : inactiveLeftIconRef;
    const leftIconTimeout = {
        appear: transitions.duration.shorter,
        enter: transitions.duration.shorter,
        exit: transitions.duration.shorter,
    };
    // eslint-disable-next-line
    const addLeftIconEndListener = (done: any) => {
        leftIconRef.current?.addEventListener(
            'transitionend',
            done,
            false,
        );
    };

    const params = {
        variant,
        size,
        active: Boolean(active),
        elevation: Boolean(elevation),
        disabled: Boolean(disabled),
        disableFocus: Boolean(disableFocus),
        leftIconExists,
        leftIconColor,
        rotateRightIcon,
        rightIconExists,
    };

    const styles = useStyles({ ...params, ...styleProps });

    if (hidden) { return null; }

    return (
        <button
            ref={ref}
            type="button"
            className={cn('root', params)}
            css={styles.root}
            disabled={disabled}
            tabIndex={focusable(!disableFocus)}
            {...restProps}
        >
            <If condition={leftIconExists}>
                <SwitchTransition mode="out-in">
                    <Transition
                        key={leftIconId}
                        nodeRef={leftIconRef}
                        timeout={leftIconTimeout}
                        addEndListener={addLeftIconEndListener}
                        enter
                    >
                        <div
                            ref={leftIconRef}
                            className={cn('leftIcon')}
                            css={styles.leftIcon}
                        >
                            {LeftIcon}
                        </div>
                    </Transition>
                </SwitchTransition>
            </If>
            <span>{children}</span>
            <If condition={rightIconExists}>
                <div
                    className={cn('rightIcon')}
                    css={styles.rightIcon}
                >
                    {RightIcon}
                </div>
            </If>
        </button>
    );
});
