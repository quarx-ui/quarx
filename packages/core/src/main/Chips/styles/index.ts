import { KeysFromUseStyles, makeStyles } from '@core/styles';
import { cssVar } from '@core/utils/cssVars';
import { paramsToCss } from '@core/utils/paramsToCss';
import { QX_SIZE } from '@core';
import { CHIPS_VARIANT } from '@core/src/main/Chips/constants';
import { baseFocusStyles, stylesWithFocus } from '@core/styles/mixins/focus';
import { ChipsCSSVarKeys } from './vars';
import { ChipsStyleParams } from './types';

export const useStyles = makeStyles((
    { palette, elevations, transitions, typography },
    {
        elevation,
        active,
        size,
        leftIconColor,
        disabled,
        disableFocus,
        variant,
        rotateRightIcon,
        onlyStateIcon,
    }: ChipsStyleParams,
    {
        cssBorderWidth,
        cssFocusColor,
        cssFocusWidth,
        cssOutsideMargin,
        cssInsideMargin,
        cssPaddingX,
        cssPaddingY,
    }: Record<ChipsCSSVarKeys, string>,
) => {
    const canChangeLeftIconColor = !disabled && leftIconColor;

    const notAllowed = {
        cursor: 'not-allowed',
    };

    const press = {
        '&:active': !disabled && {
            backgroundColor: palette.colors.brand.alpha[16],
        },
    };

    const elevationOn = {
        [cssBorderWidth]: '0px',
        ...elevations.main.small,
        '&:hover': !disabled && {
            ...elevations.main.medium,
            backgroundColor: palette.colors.brand.alpha[8],
        },
        ...press,
        '&:focus-visible': !disabled && !disableFocus && {
            borderColor: 'transparent',
            ...elevations.main.medium,
            ...stylesWithFocus({
                borderColor: cssVar(cssFocusColor),
            }),
        },
    };

    const disabledAndElevation = {
        ...disabled && {
            ...notAllowed,
            borderColor: 'transparent',
            color: palette.disabled.secondary,
            boxShadow: 'none',
            backgroundColor: palette.background.main,
        },
    };

    const activeElevationState = {
        borderColor: palette.colors.brand.border,
        backgroundColor: palette.colors.brand.surface,
        '&:focus-visible': !disabled && !disableFocus && {
            ...elevations.main.medium,
            borderColor: 'transparent',
            backgroundColor: palette.colors.brand.bg,
            ...press,
            ...stylesWithFocus({
                borderColor: cssVar(cssFocusColor),
            }),
        },
    };

    const inactiveElevationState = {
        borderColor: 'transparent',
        backgroundColor: palette.background.main,
    };

    const isFilterTypeChips = variant === CHIPS_VARIANT.filter;
    const rotateActiveRightistFilterIcon = rotateRightIcon
        && active
        && isFilterTypeChips;

    const elevationOff = {
        borderColor: palette.colors.brand.border,
        backgroundColor: 'transparent',
        '&:hover': !disabled && {
            backgroundColor: palette.colors.brand.alpha[8],
        },
        ...press,
        '&:focus-visible': !disabled && !disableFocus && {
            ...elevations.main.medium,
            borderColor: 'transparent',
            ...stylesWithFocus({
                borderColor: cssVar(cssFocusColor),
            }),
        },
        ...disabled && {
            ...notAllowed,
            color: palette.disabled.secondary,
            borderColor: palette.disabled.secondary,
        },
    };

    const calcPaddingY = `calc(${cssVar(cssPaddingY)} - ${cssVar(cssBorderWidth)})`;
    const calcPaddingX = `calc(${cssVar(cssPaddingX)} - ${cssVar(cssBorderWidth)})`;

    return ({
        root: [
            {
                boxSizing: 'border-box',
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                margin: 0,
                outline: 'none',
                cursor: 'pointer',
                WebkitTapHighlightColor: 'transparent',
                borderStyle: 'solid',
                borderWidth: cssVar(cssBorderWidth),
                borderRadius: 20,
                color: palette.text.main,
                padding: `${calcPaddingY} ${calcPaddingX}`,
                transition: transitions.create([
                    'background-color',
                    'color',
                    'box-shadow',
                    'border-color',
                    'padding',
                ], { duration: transitions.duration.shorter }),

                [cssInsideMargin]: '8px',
                [cssBorderWidth]: '1px',
                [cssFocusWidth]: '3px',
                [cssFocusColor]: palette.border.focus.main,
                [cssOutsideMargin]: `calc(12px - ${cssVar(cssPaddingX)})`,
            },
            baseFocusStyles({
                transitions,
                borderWidth: cssVar(cssBorderWidth),
                focusWidth: cssVar(cssFocusWidth),
            }),
            elevation && elevationOn,
            elevation && paramsToCss(`active-${active}`)({
                'active-true': activeElevationState,
                'active-false': inactiveElevationState,
            }),
            elevation && disabledAndElevation,
            !elevation && elevationOff,
            paramsToCss(size)({
                [QX_SIZE.medium]: {
                    [cssPaddingX]: '20px',
                    [cssPaddingY]: '8px',
                    ...typography.base.text.large,
                    fontWeight: 500,
                },
                [QX_SIZE.small]: {
                    [cssPaddingX]: '16px',
                    [cssPaddingY]: '6px',
                    ...typography.base.text.medium,
                    fontWeight: 500,
                },
            }),
        ],

        content: {},

        leftIcon: [
            {
                display: 'flex',
                marginRight: cssVar(cssInsideMargin),
                marginLeft: cssVar(cssOutsideMargin),
            },
            canChangeLeftIconColor && {
                color: leftIconColor,
            },
            onlyStateIcon && {
                width: 24,
                transition: transitions.create(['width', 'padding', 'margin', 'opacity']),
            },
            onlyStateIcon && !active && {
                width: 0,
                padding: 0,
                margin: 0,
                opacity: 0,
            },
        ],

        rightIcon: [
            {
                display: 'flex',
                marginLeft: cssVar(cssInsideMargin),
                marginRight: cssVar(cssOutsideMargin),
                transition: transitions.create('transform', {
                    duration: transitions.duration.shorter,
                }),
            },
            rotateActiveRightistFilterIcon && {
                transform: 'rotate(180deg)',
            },
        ],
    });
});

export type ChipsStyleKeys = KeysFromUseStyles<typeof useStyles>;
export * from './types';
export * from './vars';
