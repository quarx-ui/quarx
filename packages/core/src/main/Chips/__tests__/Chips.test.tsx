import React, { ReactElement } from 'react';
import { render, screen } from '@testing-library/react';
import { expectPropsMapInClasses, testStyleParams } from '@core/test-utils';
import { ChipsProps, ChipsStyleParams, QX_SIZE, CHIPS_VARIANT } from '@core';
import { ClosingIcon } from '@core/src/main/Chips/assets';
import { Chips } from '..';

describe('Chips snapshots', () => {
    Chips.displayName = 'Chips';

    const bools = [false, true];
    const defaultStylesProps = {
        size: QX_SIZE.medium,
        active: false,
        disableFocus: false,
        variant: CHIPS_VARIANT.input,
        disabled: false,
        elevation: false,
        leftIconExists: false,
        rightIconExists: false,
        rotateRightIcon: false,
    };
    testStyleParams<ChipsStyleParams, ChipsProps>(
        Chips,
        {
            size: QX_SIZE.medium,
            active: false,
            disableFocus: false,
            variant: CHIPS_VARIANT.input,
            disabled: false,
            elevation: false,
            leftIconExists: false,
            rightIconExists: false,
            rotateRightIcon: false,
        },
        { children: 'chips' },
    )({
        leftIconExists: [false],
        leftIconColor: ['green'],
        rightIconExists: [false],
        rotateRightIcon: bools,
        variant: Object.values(CHIPS_VARIANT),
        active: bools,
        disabled: bools,
        elevation: bools,
        disableFocus: bools,
        size: [QX_SIZE.small, QX_SIZE.medium],
    });

    const testIcons = (propName: string, styleParam: string, value: ReactElement | boolean) => (
        (): void => {
            const testId = `testStyleParams-${styleParam}-${value}`;

            const testProps = { [propName]: value };
            const testStyleProps = { [styleParam]: Boolean(value) };
            const { asFragment } = render(
                <Chips
                    {...defaultStylesProps}
                    {...testProps}
                    data-testid={testId}
                />,
            );

            const element = screen.getByTestId(testId);
            expectPropsMapInClasses(element)(testStyleProps);
            expect(asFragment()).toMatchSnapshot();
        }
    );

    const Icon = ClosingIcon.small;
    it('style params exists left icon', testIcons(
        'leftIcon',
        'leftIconExists',
        <Icon />,
    ));
    it('style params exists right icon', testIcons(
        'rightIcon',
        'rightIconExists',
        <Icon />,
    ));
});

describe('Chips behavior', () => {
    it('text should be in the document', () => {
        render(<Chips>Chips</Chips>);

        expect(screen.queryByText('Chips')).toBeInTheDocument();
    });
});
