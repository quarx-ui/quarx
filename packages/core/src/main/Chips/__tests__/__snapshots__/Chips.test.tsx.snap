// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chips snapshots style params active false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-active-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params active true 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_active QxChips_leftIconExists QxChips_rotateRightIcon css-8vp9tz-QxChips-root"
    data-testid="testStyleParams-active-true"
    tabindex="0"
    type="button"
  >
    <div
      class="QxChips-leftIcon css-3lolv3-QxChips-leftIcon"
      style="opacity: 1; transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.29289 12.2901C6.68342 11.8996 7.31658 11.8996 7.70711 12.2901L9.43724 14.0202C9.70392 14.2868 9.83727 14.4202 10.003 14.4202C10.1686 14.4202 10.302 14.2868 10.5686 14.0201L17.2902 7.29782C17.6807 6.90727 18.3138 6.90723 18.7044 7.29774C19.0949 7.68824 19.095 8.3214 18.7045 8.71195L10.7101 16.7072C10.5226 16.8947 10.2683 17.0001 10.003 17.0001C9.7378 17.0001 9.48344 16.8948 9.29589 16.7072L6.29289 13.7043C5.90237 13.3138 5.90237 12.6807 6.29289 12.2901Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params disableFocus false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-disableFocus-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params disableFocus true 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_disableFocus QxChips_rotateRightIcon css-1rwg5gq-QxChips-root"
    data-testid="testStyleParams-disableFocus-true"
    tabindex="-1"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params disabled false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-disabled-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params disabled true 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_disabled QxChips_rotateRightIcon css-zj0nod-QxChips-root"
    data-testid="testStyleParams-disabled-true"
    disabled=""
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params elevation false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-elevation-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params elevation true 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_elevation QxChips_rotateRightIcon css-1yl5uw3-QxChips-root"
    data-testid="testStyleParams-elevation-true"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params exists left icon 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_leftIconExists css-8vp9tz-QxChips-root"
    data-testid="testStyleParams-leftIconExists-[object Object]"
    tabindex="0"
    type="button"
  >
    <div
      class="QxChips-leftIcon css-3lolv3-QxChips-leftIcon"
      style="opacity: 1; transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.7761 4.27733C13.0688 3.98468 13.0688 3.51019 12.7761 3.21754C12.4835 2.92488 12.009 2.92488 11.7163 3.21754L8.42058 6.51328C8.22074 6.71312 8.12082 6.81304 7.99666 6.81304C7.87249 6.81304 7.77258 6.71312 7.57274 6.51328L4.27602 3.21656C3.98336 2.92391 3.50888 2.92391 3.21622 3.21656C2.92357 3.50922 2.92357 3.9837 3.21622 4.27636L6.51294 7.57308C6.71278 7.77291 6.8127 7.87283 6.8127 7.99699C6.8127 8.12116 6.71278 8.22108 6.51294 8.42091L3.21607 11.7178C2.92342 12.0104 2.92342 12.4849 3.21607 12.7776C3.50873 13.0702 3.98321 13.0702 4.27587 12.7776L7.57274 9.48071C7.77258 9.28087 7.87249 9.18095 7.99666 9.18095C8.12082 9.18095 8.22074 9.28087 8.42058 9.48071L11.7165 12.7766C12.0091 13.0693 12.4836 13.0693 12.7763 12.7766C13.0689 12.4839 13.0689 12.0095 12.7763 11.7168L9.48037 8.42091C9.28053 8.22108 9.18062 8.12116 9.18062 7.99699C9.18062 7.87283 9.28053 7.77291 9.48037 7.57308L12.7761 4.27733Z"
        />
      </svg>
    </div>
    <span />
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params exists right icon 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rightIconExists css-orbi1x-QxChips-root"
    data-testid="testStyleParams-rightIconExists-[object Object]"
    tabindex="0"
    type="button"
  >
    <span />
    <div
      class="QxChips-rightIcon css-1rlt4cc-QxChips-rightIcon"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.7761 4.27733C13.0688 3.98468 13.0688 3.51019 12.7761 3.21754C12.4835 2.92488 12.009 2.92488 11.7163 3.21754L8.42058 6.51328C8.22074 6.71312 8.12082 6.81304 7.99666 6.81304C7.87249 6.81304 7.77258 6.71312 7.57274 6.51328L4.27602 3.21656C3.98336 2.92391 3.50888 2.92391 3.21622 3.21656C2.92357 3.50922 2.92357 3.9837 3.21622 4.27636L6.51294 7.57308C6.71278 7.77291 6.8127 7.87283 6.8127 7.99699C6.8127 8.12116 6.71278 8.22108 6.51294 8.42091L3.21607 11.7178C2.92342 12.0104 2.92342 12.4849 3.21607 12.7776C3.50873 13.0702 3.98321 13.0702 4.27587 12.7776L7.57274 9.48071C7.77258 9.28087 7.87249 9.18095 7.99666 9.18095C8.12082 9.18095 8.22074 9.28087 8.42058 9.48071L11.7165 12.7766C12.0091 13.0693 12.4836 13.0693 12.7763 12.7766C13.0689 12.4839 13.0689 12.0095 12.7763 11.7168L9.48037 8.42091C9.28053 8.22108 9.18062 8.12116 9.18062 7.99699C9.18062 7.87283 9.28053 7.77291 9.48037 7.57308L12.7761 4.27733Z"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params leftIconColor green 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_leftIconColor_green QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-leftIconColor-green"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params leftIconExists false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-leftIconExists-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params rightIconExists false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-rightIconExists-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params rotateRightIcon false 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-rotateRightIcon-false"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params rotateRightIcon true 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-rotateRightIcon-true"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params size medium 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-size-medium"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params size small 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_small QxChips_rotateRightIcon css-y4x4t8-QxChips-root"
    data-testid="testStyleParams-size-small"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params variant filter 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_filter QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-variant-filter"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;

exports[`Chips snapshots style params variant input 1`] = `
<DocumentFragment>
  <button
    class="QxChips QxChips_variant_input QxChips_size_medium QxChips_rotateRightIcon css-12yo2hf-QxChips-root"
    data-testid="testStyleParams-variant-input"
    tabindex="0"
    type="button"
  >
    <span>
      chips
    </span>
  </button>
</DocumentFragment>
`;
